.docs/contributing-book/src/documentation/project-quality/documentation/code/comments.md

```diff
@@ -0,0 +1,16 @@
+# Comments
+
+Comments are used by developers for themselves or other developers to provide insight into some functionality.
+
+There are many ways to achieve the same outcome for a line of code however there are implementation tradeoffs to consider and a developer might be interested in knowing why the current approach has been chosen. 
+
+Moreover, it may not be immediately clear why, or what, some line of code is doing so it may be a good idea to add a comment summarizing the intent behind the implementation.
+
+The following snippet looks at two items being documented using the comment syntax `//`.
+
+- `Item1` has poor comments that do not convey any meaningful information and it's better to not include them at all.
+- `Item2` has taken the approach of describing the context in order to provide meaning behind each field
+
+```rust
+{{#include ../../../../code/bad_documentation/src/lib.sw:3:19}}
```

Comment:
I haven't looked into anchors so I don't know if they can be auto generated. I assume that you must write all of that into the file yourself therefore I have opted for this to keep the code cleaner and the docs.

Reply:
[Here](https://github.com/FuelLabs/sway/blob/ac745eaca9b4d42e6e216028f429453f8c3f467b/docs/src/basics/blockchain_types.md?plain=1#L54) is an example of where I used anchors in the Sway book and the corresponding [anchor](https://github.com/FuelLabs/sway/blob/ac745eaca9b4d42e6e216028f429453f8c3f467b/sway-lib-std/src/identity.sw#L7)

Reply:
Right, I have seen them before however I am unconvinced that it's worth the effort. The argument I see to use them is that if the file changes, changes to the code or formatting, then the book documentation will need to be update rather than just the code.
Again, I'm unsure if it's worth the effort of putting these throughout the files.

I'm open to it and now that I've realized that I cannot think of a counter to my argument I will probably do this... man, autism is meant to make this easier and not trigger my perfectionism...